src/ShallowTraversal.js

```diff
@@ -127,9 +133,16 @@ export function buildPredicate(selector) {
       }
       break;
 
+    case 'object':
+      if (!Array.isArray(selector) && selector !== null && !isEmpty(selector)) {
+        return node => nodeMatchesObjectProps(node, selector);
+      }
+      throw new TypeError(
+        'Enzyme::Selector does not support an empty object, array, or null as a selector'
+      );
 
     default:
-      throw new TypeError('Expecting a string or Component Constructor');
+      throw new TypeError(`Enzyme::Selector expects a string, object, or Component Constructor`);
```

Comment:
ah yes i misread the wording. it doesn't support null, or an array, or an empty object - i read "empty" as applying to all three. perhaps reverse the list on line 141?